SYNC.md

```diff
@@ -0,0 +1,143 @@
+# Automerge Sync Protocol Documentation
+
+The automerge network sync protocol is designed to bring two documents into sync by exchanging messages between peers until both documents they have the same contents. It does this by exchanging rounds of sync messages. These sync messages contain two parts: 
+ * a lossily-compressed list of changes it already has (implicitly requesting the remainder)
+ * changes it believe the other peer needs
+
+Upon receiving a message, the developer calls `receiveSyncMessage(doc, syncState, message)`. This causes Automerge to apply any changes it received, then updates some internal metadata about the other peer and is ready to produce a reply. The developer can now call `generateSyncMessage(doc, syncState)` to produce the next message to a peer. These two functions are the main two functions and can continue to be called iteratively until `generateSyncMessage()` returns a `null` value indicating both documents are synchronized.
+
+## How to use the Sync Protocol
+
+Automerge synchronization occurs at a per-document level. Most automerge-based applications will be built around more than one document, so in our example code here we will assume these documents are identified by a string `docId`.
+
+Throughout the example code below we're going to assume a couple of global variables exist, described here:
+
+```
+// global variables (but maybe don't use global variables)
+const syncStates = {} // a hash of [source][docId] containing in-memory sync states
+const backends = {} // a hash by [docId] of current backend values
+```
+
+### Connecting: `loadSyncState()` or `initSyncState()`
+
+When a peer is discovered, first create a new `syncState` with `initSyncState()` and store the result somewhere associated with that peer. These `syncState` objects can be persisted between program executions as an optimization, but it is not required. All subsequent sync operations with that peer will produce a new `syncState` to replace the previous one.
+
+If you've already seen a peer, you should load your old `syncState` for them via `loadSyncState()`. This is not strictly necessary, but will reduce unnecessary computation and network traffic.
```

Comment:
`loadSyncState` isn't explained here, I presume it loads from the `syncStates` global but could maybe explain.